engine/minskyTensorOps.cc

```diff
@@ -186,8 +194,10 @@ namespace minsky
   TensorOpFactory::TensorOpFactory()
   {
     registerType<TimeOp>(OperationType::time);
-    registerOps<MultiWireBinOp, OperationType::add, OperationType::log>(*this);
+    registerOps<MultiWireBinOp, OperationType::add, OperationType::log>(*this); 
     registerOps<TensorBinOp, OperationType::log, OperationType::copy>(*this);
+    registerType<ConstOp>(OperationType::euler);
+    registerType<ConstOp>(OperationType::pi);    
```

Comment:
This will cause the factory to generate the same object (a ConstOp), regardless of the constant. This would be an error, if it weren't for the fact that these operations are scalar, and hence the tensor factory will not be called anyway.